src/client/api/pythonApi.ts

```diff
@@ -214,20 +216,42 @@ export class PythonInstaller implements IPythonInstaller {
     }
     constructor(@inject(IPythonApiProvider) private readonly apiProvider: IPythonApiProvider) {}
 
-    public install(
+    public async install(
         product: Product,
         resource?: InterpreterUri,
         cancel?: CancellationToken
     ): Promise<InstallerResponse> {
-        return this.apiProvider
-            .getApi()
-            .then((api) => api.install(ProductMapping[product], resource, cancel))
-            .then((result) => {
-                if (result === InstallerResponse.Installed) {
-                    this._onInstalled.fire({ product, resource });
-                }
-                return result;
+        let action: 'installed' | 'failed' | 'disabled' | 'ignored' = 'installed';
+        try {
+            const api = await this.apiProvider.getApi();
+            const result = await api.install(ProductMapping[product], resource, cancel);
+            if (result === InstallerResponse.Installed) {
```

Comment:
We didn't know whether users were presented with prompts to install the missing ipykernel & didn't know whether users installed or not.
I.e. after messages such as `ipykernel not installed` is logged, i still messages indicating cells are being run again, but no messages to indicate whether the kernel was installed or even if the prompt was dismissed.

This will tell us whether a kernel is install or not after the initial prompt.